sw/host/opentitanlib/src/test_utils/load_sram_program.rs

```diff
@@ -96,17 +107,52 @@ pub enum LoadSramProgramError {
 }
 impl_serializable_error!(LoadSramProgramError);
 
-/// Load a program into SRAM using JTAG.
-pub fn load_sram_program(jtag: &Rc<dyn Jtag>, vmem: &Vmem, addr: u32) -> Result<()> {
+/// Load a program into SRAM using JTAG (VMEM files)
+pub fn load_vmem_sram_program(jtag: &Rc<dyn Jtag>, vmem_filename: &PathBuf, load_addr: u32) -> Result<u32> {
+    log::info!("Loading VMEM file {}", vmem_filename.display());
+    let vmem_content = fs::read_to_string(vmem_filename)?;
+    let mut vmem = Vmem::from_str(&vmem_content)?;
+    vmem.merge_sections();
+    log::info!("Uploading program to SRAM at {:x}", load_addr);
     for section in vmem.sections() {
         log::info!(
             "Load {} words at address {:x}",
             section.data.len(),
-            addr + section.addr
+            load_addr + section.addr
         );
-        jtag.write_memory32(addr + section.addr, &section.data)?;
+        jtag.write_memory32(load_addr + section.addr, &section.data)?;
+    }
+    Ok(load_addr)
+}
+
+/// Load a program into SRAM using JTAG (ELF files)
+pub fn load_elf_sram_program(jtag: &Rc<dyn Jtag>, elf_filename: &PathBuf) -> Result<u32> {
+    log::info!("Loading ELF file {}", elf_filename.display());
+    let file_data = std::fs::read(elf_filename)
+        .context(format!("Could not read ELF file {}.", elf_filename.display()))?;
+    let file = object::File::parse(&*file_data).context(format!(
```

Comment:
This does formatting even when there is no error. Maybe `with_context(|| format!(...))`?